source_code/src/OLEDMINI/bitstreammini.c

```diff
@@ -45,38 +45,36 @@ void miniBistreamInit(bitstream_mini_t* bs, uint8_t height, uint16_t width, uint
     bs->width = width;
     bs->height = height;
     bs->dataCounter = 0;
-    bs->bufferInd = sizeof(bs->buffer);
-    bs->dataSize = (uint16_t)width * (((uint16_t)height+7) >> BITSTREAM_PIXELS_PER_BYTE_BITSHIFT);
+    bs->dataSize = (uint16_t)width * (((uint16_t)height+7) / BITSTREAM_PIXELS_PER_BYTE);
 }
 
 /*! \fn     bsGetNextByte(bitstream_mini_t* bs)
  *  \brief  Return the next data byte from flash
  *  \param  bs      pointer to initialized bitstream context to get the next word from
  *  \return next data word, or 0 if end of data reached
  */
-uint8_t miniBistreamGetNextByte(bitstream_mini_t* bs)
-{
-    // Are you requesting bytes when you've already read everything?
-    if (bs->dataCounter < bs->dataSize) 
-    {
-        bs->dataCounter++;
-        
-        // Check if we need to fetch new data from the SPI flash
-        if(bs->bufferInd >= sizeof(bs->buffer))
-        {
-            // Fetch new data from external flash
-            flashRawRead(bs->buffer, bs->addr, sizeof(bs->buffer));
-            bs->addr += sizeof(bs->buffer);
-            bs->bufferInd = 0;
-            //usbPrintf_P(PSTR("bistream buffer: %02x %02x %02x %02x %02x %02x"), bs->buffer[0], bs->buffer[1], bs->buffer[2], bs->buffer[3], bs->buffer[4], bs->buffer[5]);
-        }
-        
-         return bs->buffer[bs->bufferInd++];
-    }
-    else
-    {
-        return 0;
-    }
-}
+ uint8_t miniBistreamGetNextByte(bitstream_mini_t* bs)
+ {
+     // Are you requesting bytes when you've already read everything?
+     if (bs->dataCounter < bs->dataSize)
+     {
+         uint16_t buffer_index = bs->dataCounter % sizeof(bs->buffer);
```

Comment:
There is not a single reason why it would now take longer:
It will if BITSTREAM_BUFFER_SIZE isn't a power of 2... and are you actually 100% the compiler optimizes it?

As I said the % results in & 0x0F. You can simply check that by doing a sha256sum of the hexfile. It is absolutely equal and optimized by the compiler.:
Have you tried it?

And the main menu is the only place to see the difference: no, everytime we need to display a picture that has hundreds of pixels.

"Those are also not required and only give additional overhead." : why is that? do you actually have measurements to back your claims?

Saying that it's fine because your general feeling is that it isn't slowing down **is not the way to go**. Any modification made to this **critical code** will only be approved with proper measurements like here: https://github.com/limpkin/mooltipass/blob/master/source_code/src/OLEDMINI/oledmini.c#L344